### PR TITLE
Options

### PR DESCRIPTION
**Pull request by kmuzalewska**

### CODE REVIEW

File: JPetOptionValidator/JPetOptionValidator.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  fValidatorMap = generateValidationMap();
+}
+
+bool JPetOptionValidator::areCorrectOptions(const std::map<std::string, boost::any>& optionsMap, std::vector<std::string>& isOption)
+{
+  //auto validationMap = generateValidationMap();
+  for (auto & checkGroup : fValidatorMap) {
+    if (std::find(isOption.begin(), isOption.end(), checkGroup.first ) != isOption.end()) {
+      for (auto & checkFunc : checkGroup.second) {
+  //       std::cout<<"areCorrectOptions: "<<checkGroup.first<<std::endl;
```
Comment (wkrzemien) [inline JPetOptionValidator/JPetOptionValidator.cpp]: remove

File: JPetOptions/JPetOptionsTools.cpp [hunk 2]
```diff
@@ -39,17 +42,64 @@ bool createConfigFileFromOptions(const Options& options, const std::string& outF
...
 {
   pt::ptree optionsTree;
-  Options mapOptions, emptyMap;
+  std::map<std::string, boost::any> mapOptions, emptyMap;
   if (JPetCommonTools::ifFileExisting(filename)) {
     try {
       pt::read_json(filename, optionsTree);
+      //std::vector<std::string> allowedTypes = { "int", "std::string", "bool", "std::vector<std::string>", "std::vector<int>"};
+      JPetOptionsTypeHandler typeHandler(kFileWithAllowedOptionType);
+     // std::cout << "After creating object of typeHandler"<< std::endl;
```
Comment (wkrzemien) [inline JPetOptions/JPetOptionsTools.cpp]: to remove

File: JPetOptions/JPetOptionsTools.cpp [hunk 3]
```diff
@@ -39,17 +42,64 @@ bool createConfigFileFromOptions(const Options& options, const std::string& outF
...
-Options createOptionsFromConfigFile(const std::string& filename)
+std::map<std::string, boost::any> createOptionsFromConfigFile(const std::string& filename)
 {
   pt::ptree optionsTree;
-  Options mapOptions, emptyMap;
+  std::map<std::string, boost::any> mapOptions, emptyMap;
   if (JPetCommonTools::ifFileExisting(filename)) {
     try {
       pt::read_json(filename, optionsTree);
+      //std::vector<std::string> allowedTypes = { "int", "std::string", "bool", "std::vector<std::string>", "std::vector<int>"};
```
Comment (wkrzemien) [inline JPetOptions/JPetOptionsTools.cpp]: to remove

File: JPetOptions/JPetOptionsTools.cpp [hunk 4]
```diff
@@ -39,17 +42,64 @@ bool createConfigFileFromOptions(const Options& options, const std::string& outF
...
+      JPetOptionsTypeHandler typeHandler(kFileWithAllowedOptionType);
+     // std::cout << "After creating object of typeHandler"<< std::endl;
       for (auto & item : optionsTree) {
+      //  std::cout << "Option tree: "<< item.first << std::endl;
```
Comment (wkrzemien) [inline JPetOptions/JPetOptionsTools.cpp]: to remove

**Comment by wkrzemien** [comment]
Ok. Now @alekgajos  should review it.

File: JPetManager/JPetManagerTest.cpp [hunk 5]
```diff
@@ -15,8 +17,27 @@
 // void parseCmdLine(int argc, char** argv);
 // inline std::vector<JPetOptions> getOptions() const { return fOptions;}
              
+
+
+char* convertStringToCharP(const std::string& s)
+{
+  char* pc = new char[s.size() + 1];
```
Comment (wkrzemien) [inline JPetManager/JPetManagerTest.cpp]: fixed in the new PR

File: JPetCmdParser/JPetCmdParserTest.cpp [hunk 6]
```diff
@@ -43,14 +46,15 @@ BOOST_AUTO_TEST_SUITE(FirstSuite)
 
 BOOST_AUTO_TEST_CASE( parsing_1 )
 {
+  std::cout<<"parsing_1 "<<std::endl;
```
Comment (wkrzemien) [inline JPetCmdParser/JPetCmdParserTest.cpp]: fixed in next pr

File: JPetCmdParser/JPetCmdParser.cpp [hunk 7]
```diff
@@ -18,32 +18,35 @@
...
-  ("outputPath,o", po::value<std::string>(), "Location to which the outputFiles will be saved.")
-  ("range,r", po::value< std::vector<int> >()->multitoken()->default_value({ -1, -1}, ""), "Range of events to process e.g. -r 1 1000 .")
-  ("unpackerConfigFile,p", po::value<std::string>(), "xml file with TRB settings used by the unpacker program.")
-  ("unpackerCalibFile,c", po::value<std::string>(), "ROOT file with TRB calibration used by the unpacker program.")
-  ("runId,i", po::value<int>(), "Run id.")
-  ("progressBar,b", po::bool_switch()->default_value(false), "Progress bar.")
-  ("localDB,l", po::value<std::string>(), "The file to use as the parameter database.")
-  ("localDBCreate,L", po::value<std::string>(), "File name to which the parameter database will be saved.")
-  ("userCfg,u", po::value<std::string>(), "Json file with optional user parameters.");
+  ("type_std::string,t", po::value<std::string>()->required()->implicit_value(""), "Type of file: hld, zip, root or scope.")
```
Comment (wkrzemien) [inline JPetCmdParser/JPetCmdParser.cpp]: I agree. I will change it

File: JPetOptionsGenerator/JPetOptionsGeneratorTest.cpp [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+#define BOOST_TEST_MODULE JPetOptionsGeneratorTest
+#include <boost/test/unit_test.hpp>
+#include <cstdlib>
+#include "../JPetCmdParser/JPetCmdParser.h"
+#include "../JPetOptionsGenerator/JPetOptionsGenerator.h"
+using boost::any_cast;
+using namespace std;
+
+
+char* convertStringToCharP(const std::string& s)
```
Comment (wkrzemien) [inline JPetOptionsGenerator/JPetOptionsGeneratorTest.cpp]: already moved to JPetCommonTools

File: JPetManager/JPetManager.cpp [hunk 9]
```diff
@@ -94,36 +99,3 @@ void JPetManager::registerTask(const TaskGenerator& taskGen)
...
-    if(fOptions.at(0).getRunNumber() >= 0){ // if run number is not default -1
-      if(!fOptions.at(0).isLocalDB()){ // unless local DB file was provided
-	isDBrequired = true;
-      }
-    }
-  }
-
-  if(isDBrequired){
-    INFO("Attempting to set up connection to the database.");
-    DB::SERVICES::DBHandler::createDBConnection(configFilePath);
```
Comment (wkrzemien) [inline JPetManager/JPetManager.cpp]: Fixed in the next PR

File: JPetTaskInterface/JPetTaskInterface.h [hunk 10]
```diff
@@ -18,14 +18,14 @@
...
-//#include "../JPetOptionsInterface/JPetOptionsInterface.h"
+#include <boost/any.hpp>
 
 class JPetParamManager;
 
 class JPetTaskInterface
 {
 public:
-  typedef std::map<std::string, std::string> Options;
+  typedef std::map<std::string, boost::any> Options;
```
Comment (wkrzemien) [inline JPetTaskInterface/JPetTaskInterface.h]: Ok I will change it anyway to some general type that can be overloaded in the future if needed

File: JPetTaskChainExecutor/JPetTaskChainExecutor.cpp [hunk 11]
```diff
@@ -71,45 +65,37 @@ bool JPetTaskChainExecutor::process()
...
+      auto outPath  = any_cast<std::string>(currOpts.at("outputPath_std::string"));
       if (!outPath.empty()) {
-        currOpts.at("inputFile") = outPath + JPetCommonTools::extractPathFromFile(currOpts.at("inputFile")) + JPetCommonTools::extractFileNameFromFullPath(currOpts.at("inputFile"));
+        currOpts.at("inputFile_std::string") = outPath + JPetCommonTools::extractPathFromFile(any_cast<std::string>(currOpts.at("inputFile_std::string"))) + JPetCommonTools::extractFileNameFromFullPath(any_cast<std::string>(currOpts.at("inputFile_std::string")));
       }
     }
     auto taskCurr = dynamic_cast<JPetTask*> (dynamic_cast<JPetTaskLoader*>(*currentTask)->getTask());
     //auto taskCurr = std::dynamic_pointer_cast<JPetTask>((*currentTask)->getTask());
     auto taskName = taskCurr->GetName();
-    startTime = stdc::system_clock::now();
```
Comment (wkrzemien) [inline JPetTaskChainExecutor/JPetTaskChainExecutor.cpp]: It is back in the new PR

**Comment by wkrzemien** [comment]
@alekgajos : I think this PR can be closed.
And we can further concentrate on #118

**Comment by alekgajos** [comment]
Ok, I am closing this PR as the changes were moved to #118

**Pull request closed without merging**